### PR TITLE
Add some custom rule use cases and the embeditor on popular pages

### PR DESCRIPTION
**Pull request by minusworld**
Part 2 coming soon: Upgrading the `rule-ideas` page

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -7,12 +7,14 @@
...
 ---
-
 # Overview
 
-
 Semgrep is a fast, open-source, static analysis tool for finding bugs and enforcing code standards at editor, commit, or CI time. Its rules look like the code you’re searching; no more traversing abstract syntax trees or wrestling with regexes.
 
+Here's an example rule. You can try it out by clicking the "Run" button. The Semgrep rule looks like the code it matches. You can edit this rule here, try writing a rule in the [Semgrep editor](https://semgrep.dev/editor), or visit the [tutorial](https://semgrep.dev/learn).
+
+<iframe title="semgrep example no prints" src="https://semgrep.dev/embed/editor?snippet=ievans:print-to-logger" width="100%" height="432px" frameborder="0"></iframe>
```
Comment (pabloest) [inline docs/index.md:16]:
```suggestion
<iframe title="Semgrep example no prints" src="https://semgrep.dev/embed/editor?snippet=ievans:print-to-logger" width="100%" height="432px" frameborder="0"></iframe>
```

File: docs/getting-started.md [hunk 2]
```diff
@@ -74,8 +74,23 @@ $ semgrep --config p/security-audit WebGoat
...
+- Lightweight vulnerability detection
+- Scanning configuration files
+- And more! Check out more use cases [here](writing-rules/rule-ideas.md).
+
+
 Visit [Writing Rules > Getting started](writing-rules/overview.md) for an in-depth guide and reference material.
 
+This rule is used to find and discourage `print(...)` in production code. You can edit this rule here or visit the [editor](https://semgrep.dev/editor) to try writing and deploying your own rule.
+
+<iframe title="semgrep example python flask routes" src="https://semgrep.dev/embed/editor?snippet=ievans:print-to-logger" width="100%" height="432px" frameborder="0"></iframe>
```
Comment (pabloest) [inline docs/getting-started.md:92]:
```suggestion
<iframe title="Semgrep example with Python Flask routes" src="https://semgrep.dev/embed/editor?snippet=ievans:print-to-logger" width="100%" height="432px" frameborder="0"></iframe>
```

File: docs/getting-started.md [hunk 3]
```diff
@@ -74,8 +74,23 @@ $ semgrep --config p/security-audit WebGoat
...
+- Detecting API routes, database models, or similar code segments
+- Identifying authentication violations
+- Lightweight vulnerability detection
+- Scanning configuration files
+- And more! Check out more use cases [here](writing-rules/rule-ideas.md).
+
+
 Visit [Writing Rules > Getting started](writing-rules/overview.md) for an in-depth guide and reference material.
 
+This rule is used to find and discourage `print(...)` in production code. You can edit this rule here or visit the [editor](https://semgrep.dev/editor) to try writing and deploying your own rule.
```
Comment (pabloest) [inline docs/getting-started.md:90]:
```suggestion
This rule is used to find and discourage `print(...)` in production code. You can edit this rule here or visit the [editor](https://semgrep.dev/editor) to write and deploy your own rule.
```

File: docs/writing-rules/overview.md [hunk 4]
```diff
@@ -7,6 +7,17 @@ If you want the best introduction to writing Semgrep rules, use the interactive,
...
+You can write rules that do things like:
+
+- Automate code review comments
+- Identify secure coding violations
+- Scan configuration files
+- And more!
+
+This rule detects the use of `is` when comparing Python strings. `is` checks reference equality, not value equality, and can exhibit nondeterministic behavior.
+
+<iframe title="Semgrep example python is comparison" src="https://semgrep.dev/embed/editor?snippet=B47A" width="100%" height="432px" frameborder="0"></iframe>
```
Comment (pabloest) [inline docs/writing-rules/overview.md:19]:
```suggestion
<iframe title="Semgrep example Python is comparison" src="https://semgrep.dev/embed/editor?snippet=B47A" width="100%" height="432px" frameborder="0"></iframe>
```

**Review by pabloest — APPROVED** [review]
@minusworld I made some minor suggestions but overall the changes look good to me.

Do note this issue about using the embeditor though: https://github.com/returntocorp/semgrep-app/issues/2341

I worry about going too far down the path of replacing text with the embeditor before this is sorted out.

**Comment by minusworld** [comment]
> @minusworld I made some minor suggestions but overall the changes look good to me.
> 
> Do note this issue about using the embeditor though: [returntocorp/semgrep-app#2341](https://github.com/returntocorp/semgrep-app/issues/2341)
> 
> I worry about going too far down the path of replacing text with the embeditor before this is sorted out.

Hmm, yeah. Do we know if there's a way to get that content searchable?

**Comment by minusworld** [comment]
cf. https://developers.google.com/search/docs/advanced/guidelines/rich-media-files

> IFrames
> IFrames are sometimes used to display content on web pages. Content displayed via iFrames may not be indexed and available to appear in Google's search results. We recommend that you avoid the use of iFrames to display content. If you do include iFrames, make sure to provide additional text-based links to the content they display, so that Googlebot can crawl and index this content.

**Comment by pabloest** [comment]
@minusworld Not using iframes as far as I know. My guess is that the real solution is to convert it to an embeddable component that renders directly.

**Comment by minusworld** [comment]
I think those Ruby links are actually dead. I'll remove them in a future PR

**Pull request merged**